storage/replica_test.go

```diff
@@ -487,6 +487,24 @@ func TestRangeLeaderLease(t *testing.T) {
 	if held, expired := hasLease(tc.rng, tc.clock.Now()); held || !expired {
 		t.Errorf("expected another replica to have expired lease")
 	}
+
+	// Verify that command returns NotLeaderError when lease is rejected.
+	rng, err := NewReplica(testRangeDescriptor(), tc.store)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rng.proposeRaftCommandFn = func(id cmdIDKey, cmd roachpb.RaftCommand) <-chan error {
+		errChan := make(chan error, 1)
+		errChan <- &roachpb.LeaseRejectedError{
+			Message: "replica not found",
+		}
+		return errChan
+	}
+
+	err = rng.redirectOnOrAcquireLeaderLease(nil, tc.clock.Now())
+	if lErr, ok := err.(*roachpb.NotLeaderError); !ok || lErr == nil {
```

Comment:
Yeah, that's true. Still, it seems to be in alignment with our practices to avoid this check. For the curious: http://play.golang.org/p/tLd4XGxKjQ